superset-frontend/src/views/CRUD/data/database/DatabaseModal/ExtraOptions.tsx

```diff
@@ -198,6 +198,23 @@ const ExtraOptions = ({
                 />
               </div>
             </StyledInputContainer>
+            <StyledInputContainer>
+              <div className="input-container">
+                <IndeterminateCheckbox
+                  id="disable_preview_data"
+                  indeterminate={false}
+                  checked={!!db?.extra_json?.disable_preview_data}
+                  onChange={onExtraInputChange}
+                  labelText={t('Disable SQL Lab data preview queries')}
+                />
+                <InfoTooltip
+                  tooltip={t(
+                    'The disable_preview_data field is a boolean specifying whether or not data ' +
+                      'preview queries will be run when fetching table metadata in SQL Lab.',
```

Comment:
This could probably be simplified. Something along these lines :
> Disable data preview when fetching table metadata in SQL Lab. Useful to avoid browser performance issues when using databases with very wide tables.

Reply:
Implemented in most recent commit